File: Praticas Prontas/pratica 7/Conversor/principal.c
/* -- GRUPO: 
-- Pedro van Rooij Costa
-- Nelson Mariano Leite Neto
-- Bruno Freitas */

#include "local.h"

int main() {
	int opt;
	float *resp,*a;
	while ((opt = mostreMenu()) != 0) {
		if (opt == 1) {
			a = pedirParam();
			resp = paraGraus(a);
			mostreResp(opt, a, resp);
		} else if (opt == 2) {
			a = pedirParam();
			resp = paraRadianos(a);
			mostreResp(opt, a, resp);
		}
	}
	return EXIT_SUCCESS;
}

File: Praticas Prontas/Proj 3/Principal.hs
-- GRUPO: 
-- Pedro van Rooij Costa
-- Nelson Mariano Leite Neto
-- Bruno Freitas

import System (getArgs)
import System.IO
import Utilitarios
import List

main = do
	args <- getArgs
	case args of
		[fname] -> do			
			fstr <- readFile fname

			let (cabecalho, imagem) = retirarCabecalho fstr
			let largura = pegarLargura cabecalho
			let imagemSemPadding = removerPadding imagem largura
			let pixels = fazerTriplas imagemSemPadding
			-- Funcao que pega a transposta da matriz passada como parametro
			let rgb = transpose pixels
			let rgbEmLinhas = map (rearranjarPixelList largura) rgb
			let nomeInitArq = take (length fname - 4) fname
			let nomesRGB = map (nomeInitArq++)["_Red.txt","_Green.txt","_Blue.txt"]
			arquivosRGB <- sequence [openFile file WriteMode | file <- nomesRGB]

			sequence [escreverNoArquivo arq str | (arq, str) <- zip arquivosRGB rgbEmLinhas]
			sequence [hClose f | f <- arquivosRGB]

			putStr "Separacao RGB concluida com sucesso!!! Os arquivos criados sao: \n"
			putStr $ concat $ map (++ "\n") nomesRGB
	
		_	-> do
				putStr "Execute o programa criado passando como parametro o nome do arquivo a ser separado em camadas\n"
				putStr "Por exemplo: ./nomePrograma teste.bmp\n"
			
